test/e2e/gateway_single_spoke_gcp_test.go

```diff
@@ -0,0 +1,295 @@
+
+				Eventually(func(ctx SpecContext) bool {
+					dnsrecord := &mgcv1alpha1.DNSRecord{ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%s", gwGCP.Name, "https"),
+						Namespace: gwGCP.Namespace,
+					},
+					}
+					if err := tconfig.HubClient().Get(ctx, client.ObjectKeyFromObject(dnsrecord), dnsrecord); err != nil {
+						GinkgoWriter.Printf("[debug] unable to get DNSRecord: '%s'\n", err)
+						return false
+					}
+					return meta.IsStatusConditionTrue(dnsrecord.Status.Conditions, conditions.ConditionTypeReady)
+				}).WithTimeout(300 * time.Second).WithPolling(10 * time.Second).WithContext(ctx).Should(BeTrue())
+
+				// still need to wait some seconds to the dns server to actually start
+				// resolving the hostname
+				select {
+				case <-ctx.Done():
+				case <-time.After(30 * time.Second):
+				}
+
+				By("ensuring the authoritative nameserver resolves the GCP hostname")
+
+				// speed up things by using the authoritative nameserver
+				nameservers, err := net.LookupNS(tconfig.ManagedZoneGCP())
+				Expect(err).ToNot(HaveOccurred())
+				GinkgoWriter.Printf("[debug] authoritative nameserver used for DNS record resolution: %s\n", nameservers[0].Host)
+
+				authoritativeResolver := &net.Resolver{
+					PreferGo: true,
+					Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+						d := net.Dialer{Timeout: 10 * time.Second}
+						return d.DialContext(ctx, network, strings.Join([]string{nameservers[0].Host, "53"}, ":"))
+					},
+				}
+
+				Eventually(func(ctx SpecContext) bool {
+					c, cancel := context.WithTimeout(ctx, 10*time.Second)
+					defer cancel()
+					IPs, err := authoritativeResolver.LookupHost(c, string(hostnameGCP))
+					if err != nil {
+						GinkgoWriter.Printf("[debug] LooupHost error: '%s'\n", err)
+					}
+					return err == nil && len(IPs) > 0
+				}).WithTimeout(300 * time.Second).WithPolling(10 * time.Second).WithContext(ctx).Should(BeTrue())
+
+				By("performing a GET using HTTPS GCP")
+				{
+					// use the authoritative nameservers
+					dialer := &net.Dialer{Resolver: authoritativeResolver}
+					dialContext := func(ctx context.Context, network, addr string) (net.Conn, error) {
+						return dialer.DialContext(ctx, network, addr)
+					}
+					http.DefaultTransport.(*http.Transport).DialContext = dialContext
+					http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+					httpClient := &http.Client{}
+
+					var resp *http.Response
+					Eventually(func(ctx SpecContext) error {
+						resp, err = httpClient.Get("https://" + string(hostnameGCP))
+						if err != nil {
+							GinkgoWriter.Printf("[debug] GET error: '%s'\n", err)
+							return err
+						}
+						return nil
+					}).WithTimeout(300 * time.Second).WithPolling(10 * time.Second).WithContext(ctx).ShouldNot(HaveOccurred())
+
+					defer resp.Body.Close()
+					Expect(resp.StatusCode).To(Equal(http.StatusOK))
+				}
+			})
+
+		})
+
+	})
+
+})
```

Comment:
This test is out of date with the equivalent one on main. I think however we can just remove it see https://github.com/Kuadrant/multicluster-gateway-controller/commit/55ff95795d94099bca4bc46c1a63bc83af829dbd#r124848829